contents/testing/writing-integration-test.md

```diff
@@ -0,0 +1,143 @@
+<frontmatter>
+  title: Integration Testing
+  footer: footer.md
+  head: head.md
+  siteNav: mainNav.md
+  pageNav: 3
+</frontmatter>
+
+{{ navbar | safe }}
+
+<div class="website-content">
+
+# Integration Testing
+
+**Author(s): [Jacob Li PengCheng](https://github.com/jacoblipech)**
+
+Reviewers:
+
+## What is integration testing?
+
+Integration testing is a part of software testing where individual units of your application are combined and tested as a group. This usually happens after [unit testing stage](http://softwaretestingfundamentals.com/unit-testing/) and before the [system testing stage](https://www.guru99.com/system-testing.html).
+
+## Why is integration testing important to SE?
+
+In a software application, each software module is usually designed and unit tested by different programmers. Hence, with different understanding of the software requirements, integration testing is necessary to ensure that software modules work in unity and expose any faults in the interaction between different parts of the units. 
+
+<box type="info">Suppose you are in charge of the **data collection component** and it passed the **data** collected from the users to the **data analysis component**. If the **data analysis component** assumed your **data** to be of maximum 100 lines but the data you sent has no limit on the number of lines, it is then the integration test that is supposed to find this discrepancy.</box>
+
+Hence, integration testing is an important aspect which can be applied to any software such as web, mobile and desktop applications. I will be mainly focusing on integration testing in terms of the web application. Similar concepts can be adopted by other applications.
+
+:information_source: More information on instances where integration testing is important can be found on [this article](https://www.guru99.com/integration-testing.html).
+
+## How does integration testing work?
+
+Given a sample architecture diagram of your application shown:
+
+![Sample architecture diagram](integration-test/sample-architecture.png "Sample architecture diagram")
+
+Below are a list of things to follow to execute effective integration testing:
+
+1. Prepare the integration test plan.
+1. Design the test scenarios, cases and scripts.
+1. Executing the test scripts and report defects if any.
+1. Tracking and re-testing the defects.
+1. Repeat testing until all integration tests are successful.
+
+### Integration Test Plan
+Before the actual implementation of your integration tests, it is important to decide on the approach used. Using a concrete example where by you are designing a online web ordering system. Your application may be split into different modules as shown below:
```

Comment:
> Using a concrete example where by you are designing a online web ordering system.
I think you could just state that you are using the sample application described above 🙂